src/saveload/saveload.cpp

```diff
@@ -2422,6 +2424,7 @@ struct ZSTDSaveFilter : SaveFilter {
 
 /** The format for a reader/writer type of a savegame */
 struct SaveLoadFormat {
+	uint8 index;                          ///< Index of this entry.
```

Comment:
Comment should say that the index has to be unique, unless you actually intended to have multiple formats at one index in the bitmask.

Some compile time assert that index is unique would probably be nice, but I'm not sure it's possible. (Maybe with constexpr/consteval, but not sure).

Reply:
I would love some compile time asserts, but I wouldn't even know where to start :D And `consteval` is C++20, not?